src/test/java/com/karankumar/bookproject/backend/service/AuthorServiceTest.java

```diff
@@ -0,0 +1,50 @@
+package com.karankumar.bookproject.backend.service;
+
+import com.karankumar.bookproject.annotations.IntegrationTest;
+import com.karankumar.bookproject.backend.entity.Author;
+import lombok.extern.java.Log;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+import java.util.logging.Level;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@Log
+@IntegrationTest
+public class AuthorServiceTest {
+
+    private final AuthorService sut;
+
+    public AuthorServiceTest(@Autowired AuthorService sut, @Autowired BookService bookService) {
+        bookService.deleteAll();
+        sut.deleteAll();
+        this.sut = sut;
+    }
+
+    @Test
+    public void testSaveAndConfirmDuplicateNameWithDifferentId() {
```

Comment:
Can we please add one more test as below:
1. Save an author
2. Get the saved author 
3. Save it again after getting from DB
4. Check there are 2 in DB

Difference in this scenario is we are trying to save an author that has id assigned and making sure its not updated.

Reply:
ok. will do some more unit test

Reply:
@NyorJa, could you comment the method to make it clear to someone else what is being tested (i.e. the last line in Amit's comment or something similar)?